auth-jwt/Sources/App/Middleware/BasicAuthenticator.swift

```diff
@@ -15,20 +15,24 @@
 import FluentKit
 import Hummingbird
 import HummingbirdAuth
+import HummingbirdFluent
+import NIOPosix
 
-struct BasicAuthenticator: HBAsyncAuthenticator {
-    func authenticate(request: HBRequest) async throws -> User? {
+struct BasicAuthenticator<Context: HBAuthRequestContext>: HBAuthenticator {
+    let fluent: HBFluent
+
+    func authenticate(request: HBRequest, context: Context) async throws -> AuthenticatedUser? {
         // does request have basic authentication info in the "Authorization" header
-        guard let basic = request.authBasic else { return nil }
+        guard let basic = request.headers.basic else { return nil }
 
         // check if user exists in the database and then verify the entered password
         // against the one stored in the database. If it is correct then login in user
-        let user = try await User.query(on: request.db)
+        let user = try await User.query(on: self.fluent.db())
             .filter(\.$name == basic.username)
             .first()
-        // can only verify password if user has password associated with it
-        guard let passwordHash = user?.passwordHash else { return nil }
-        guard Bcrypt.verify(basic.password, hash: passwordHash) else { return nil }
-        return user
+        guard let user = user, let passwordHash = user.passwordHash else { return nil }
+        // Do Bcrypt verify on a separate thread to not block the general task executor 
+        guard try await NIOThreadPool.singleton.runIfActive({ Bcrypt.verify(basic.password, hash: passwordHash) }) else { return nil }
```

Comment:
I don't think we need to lean on a NIOThreadPool with async-await

Reply:
Yes we do, we shouldn't be blocking the global task executor.